daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala

```diff
@@ -45,12 +45,18 @@ class DecodeV1Spec
   val dummyModuleDName = DamlLf1.DottedName.newBuilder().addSegments(dummyModuleStr).build()
   val dummyModuleName = DottedName.assertFromString(dummyModuleStr)
 
+  import com.daml.lf.transaction.VersionTimeline.Implicits._
+
+  val lfVersions =
```

Comment:
What do you think about adding two functions of types
```scala
forEveryBefore, forEveryAtOrAfter: (Version) => (Version => Unit) => Unit
```
Then we could simply swap out the `forEvery`s below and avoid a lot of `if`s.

Reply:
Thanks for doing this! 🙇